ssas/blacklist_test.go

```diff
@@ -0,0 +1,51 @@
+package ssas
+
+import (
+	"github.com/jinzhu/gorm"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"testing"
+	"time"
+)
+
+type CacheEntriesTestSuite struct {
+	suite.Suite
+	db *gorm.DB
+}
+
+func (s *CacheEntriesTestSuite) SetupSuite() {
+	s.db = GetGORMDbConnection()
+	InitializeBlacklistModels()
+}
+
+func (s *CacheEntriesTestSuite) TearDownSuite() {
+	Close(s.db)
+}
+
+func (s *CacheEntriesTestSuite) TestGetUnexpiredCacheEntries() {
+	var err error
+	entryDate := time.Now().Add(time.Minute*-5).UnixNano()
+	expiration := time.Now().Add(time.Minute*5).UnixNano()
+	e1 := BlacklistEntry{Key: "key1", EntryDate: entryDate, CacheExpiration: expiration}
+	e2 := BlacklistEntry{Key: "key2", EntryDate: entryDate, CacheExpiration: expiration}
+
+	if err = s.db.Save(&e1).Error; err != nil {
+		assert.FailNow(s.T(), err.Error())
+	}
+	if err = s.db.Save(&e2).Error; err != nil {
+		assert.FailNow(s.T(), err.Error())
+	}
+
+	entries, err := GetUnexpiredBlacklistEntries()
```

Comment:
Since all the entries are unexpired, we can't be sure this method filters correctly. Similarly, do we know that we handle the case of an empty cache correctly?